File: packages/react-native-client/ios/managers/LocalCameraTracksChangedListenersManager.swift
import Foundation

protocol LocalCameraTrackChangedListener: AnyObject {
    func onLocalCameraTrackChanged()
}

class LocalCameraTracksChangedListenersManager {
    private var listeners: [LocalCameraTrackChangedListener] = []

    func add(_ listener: LocalCameraTrackChangedListener) {
        listeners.append(listener)
    }

    func remove(_ listener: LocalCameraTrackChangedListener) {
      listeners.removeAll{ $0 === listener}

    }

    func notifyListeners() {
        for listener in listeners {
            listener.onLocalCameraTrackChanged()
        }
    }
}

File: packages/react-native-client/android/src/main/java/io/fishjam/reactnative/managers/LocalCameraTracksChangedListenersManager.kt
package io.fishjam.reactnative.managers

interface LocalCameraTrackChangedListener {
  fun onLocalCameraTrackChanged()
}

class LocalCameraTracksChangedListenersManager :
  SingleListenerManager<LocalCameraTrackChangedListener>(LocalCameraTrackChangedListener::onLocalCameraTrackChanged)
